src/plugins/field_formats/common/converters/duration.ts

```diff
@@ -128,53 +127,51 @@ export class DurationFormat extends FieldFormat {
   };
 }
 
+// function to calculate the precision part of the value
+const calculatePrecision = (rawValue: number, unitValue: number, seconds: number) => {
+  return Math.floor(unitValue) + (rawValue - unitValue * seconds) / seconds;
+};
+
+// Array of units is to find the first unit duration value that is not 0
+const units = [
+  { getValue: (dur: moment.Duration) => dur.years(), seconds: 31536000, method: 'asYears' },
+  // Note: 31 days is used as a month in the duration format
```

Comment:
Based on `moment.duration({ months: 1 }).asDays()` looks like moment is using `30` instead of `31`. Should we align since we reply on moment duration here? Some calculators even use 30,417.

Reply:
I will check, that's a good question, how much days has a month? I think I've set to a different value first, but a then a unit test broke, will revisit 👍

Reply:
Yes, let's use 30 👍 done